### ISSUE TITLE
In Mill 0.11, importing multiple projects with same filename leads to ambiguous import

### ISSUE THREAD

**Issue by sequencer** [com-lihaoyi/mill#2683]
We(the Chisel guys) are using Scala for building projects from sources.
However when we imports multiple file, e.g:
```scala
import $file.someproject.common
import $file.anotherproject.common
```
These two common becomes ambiguous:
```
reference to common is ambiguous;
[error] it is imported twice in the same scope by
[error] import millbuild.someproject.common
[error] and import millbuild.anotherproject.common
```

Thanks to #2377, I'm able to see what happens, multiple `common` objects are defined as below:
```scala
package millbuild.someproject

object common extends common
class common extends _root_.mill.main.RootModule.Foreign(Some(_root_.mill.define.Segments.labels("foreign-modules", "someproject", "common")))
```
and 
```scala
package millbuild.anotherproject
object common extends common
class common extends _root_.mill.main.RootModule.Foreign(Some(_root_.mill.define.Segments.labels("foreign-modules", "anotherproject", "common"))) 
```
These two are different object under different namespaces, but when using it from the main build script needs to workaround by using `millbuild.someproject.common.SomeBuildModule`, `millbuild.anotherproject.common.AnotherBuildModule`.

**Comment by lefou** [comment]
This is expected Scala behavior. The solution is to rename at least n-1 ambiguous imports to disambiguate them.

```scala
import $file.someproject.{common => some}
import $file.anotherproject.{common => another}
```

**Comment by sequencer** [comment]
I tried this, however this seems not working.

**Comment by lihaoyi** [comment]
Going to call this fixed in 0.12.0, where we can refer to things fully qualified via `$file.someproject.common` and `$file.anotherproject.common`, so there won't necessarily be conflicts (whereas previously the `import` was required which cause the conflict). There may be new usability/functional issues in 0.12.0 but let's open new issues for those when we hit them
